libnetwork/store_test.go

```diff
@@ -5,17 +5,15 @@ import (
 	"testing"
 
 	"github.com/docker/docker/libnetwork/config"
-	store "github.com/docker/docker/libnetwork/internal/kvstore"
 	"github.com/docker/docker/libnetwork/netlabel"
 	"github.com/docker/docker/libnetwork/options"
 )
 
-func testLocalBackend(t *testing.T, provider, url string, storeConfig *store.Config) {
-	cfgOptions := []config.Option{func(c *config.Config) {
-		c.Scope.Client.Provider = provider
-		c.Scope.Client.Address = url
-		c.Scope.Client.Config = storeConfig
-	}}
+func testLocalBackend(t *testing.T, path, bucket string) {
+	cfgOptions := []config.Option{
+		config.OptionDataDir(path),
+		func(c *config.Config) { c.DatastoreBucket = bucket },
+	}
 
 	cfgOptions = append(cfgOptions, config.OptionDriverConfig("host", map[string]interface{}{
```

Comment:
Minor nit; it looks like `cfgOptions` is never conditionally modified, so we only use it as an intermediate slice. So technically this could be;


```go
testController, err := New(
    config.OptionDataDir(path),
    func(c *config.Config) { c.DatastoreBucket = bucket },
    config.OptionDriverConfig("host", map[string]interface{}{
        netlabel.GenericData: options.Generic{},
    }),
) 
````